### PR TITLE
ref(integrations): Actually check sync settings before syncing

### PR DESCRIPTION
**Pull request by macqueen**
lmk if you have suggestions for a better way to do this. i figured this is at least flexible, but will minimize what we have to do to make it work for jira and vsts?

### CODE REVIEW

**Review by mnoble — COMMENTED** [review]
Just a couple questions.

File: src/sentry/integrations/issues.py [hunk 1]
```diff
@@ -133,6 +133,24 @@ def make_external_key(self, data):
...
+    def should_sync(self, attribute):
+        try:
+            key = getattr(self, '%s_key' % attribute)
+        except AttributeError:
+            return False
+
+        if key is None:
+            return False
+
+        config = self.org_integration.config
```
[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/integrations/issues.py]: yeah we want to give people the option of saying that they do or don't want us to sync status/assignee/comments. it'll look something like this (although evan is working on redo-ing the ui, so not *exactly* like that)
screenshot 2018-07-20 11 09 25

those settings will be configured on an organization/integration level

File: src/sentry/models/groupassignee.py [hunk 2]
```diff
@@ -54,10 +54,16 @@ def sync_group_assignee_inbound(integration, email, external_issue_key, assign=T
...
+    for org_id in integration.organizations.values_list('id', flat=True):
+        installation = integration.get_installation(org_id)
+        if installation.should_sync('inbound_assignee'):
+            orgs_with_sync_enabled.append(org_id)
+
     affected_groups = list(
         Group.objects.get_groups_by_external_issue(
             integration, external_issue_key,
-        ),
+        ).filter(project__organization_id__=orgs_with_sync_enabled),
```
[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/models/groupassignee.py]: ffffff, yeah, will fix

[the remark this answers is not in the document]
Comment (macqueen) [reply src/sentry/models/groupassignee.py]: yeah there are tests that are failing
```
___________ GroupAssigneeTestCase.test_assignee_sync_inbound_assign ____________
Traceback (most recent call last):
  File "/home/travis/build/getsentry/sentry/tests/sentry/models/test_groupassignee.py", line 224, in test_assignee_sync_inbound_assign
    integration, user_no_access.email, 'APP-123'
  File "/home/travis/build/getsentry/sentry/src/sentry/models/groupassignee.py", line 66, in sync_group_assignee_inbound
    ).filter(project__organization_id__=orgs_with_sync_enabled),
  File "/home/travis/virtualenv/python2.7.13/lib/python2.7/site-packages/django/db/models/query.py", line 593, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.13/lib/python2.7/site-packages/django/db/models/query.py", line 611, in _filter_or_exclude
    clone.query.add_q(Q(*args, **kwargs))
  File "/home/travis/virtualenv/python2.7.13/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1204, in add_q
    clause = self._add_q(where_part, used_aliases)
  File "/home/travis/virtualenv/python2.7.13/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1240, in _add_q
    current_negated=current_negated)
  File "/home/travis/virtualenv/python2.7.13/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1103, in build_filter
    allow_explicit_fk=True)
  File "/home/travis/virtualenv/python2.7.13/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1363, in setup_joins
    names, opts, allow_many, allow_explicit_fk)
  File "/home/travis/virtualenv/python2.7.13/lib/python2.7/site-packages/django/db/models/sql/query.py", line 1283, in names_to_path
    "Choices are: %s" % (name, ", ".join(available)))
```
i just uhh, neglected to run them 😬

File: src/sentry/integrations/issues.py [hunk 3]
```diff
@@ -133,6 +133,24 @@ def make_external_key(self, data):
 
 
 class IssueSyncMixin(IssueBasicMixin):
+    comment_key = None
+    outbound_status_key = None
+    inbound_status_key = None
+    outbound_assignee_key = None
```
Comment (EvanPurkhiser) [inline src/sentry/integrations/issues.py]: This will change soon since it will be based on project

**Review by evanpurkhiser — APPROVED** [review]
lgtm

**Pull request merged**